rct229/rulesets/ashrae9012019/section5/section5rule41.py

```diff
@@ -0,0 +1,130 @@
+    get_opaque_surface_type,
+)
+from rct229.rulesets.ashrae9012019.ruleset_functions.get_surface_conditioning_category_dict import (
+    SurfaceConditioningCategory as SCC,
+)
+from rct229.rulesets.ashrae9012019.ruleset_functions.get_surface_conditioning_category_dict import (
+    get_surface_conditioning_category_dict,
+)
+from rct229.utils.jsonpath_utils import find_one
+
+
+class Section5Rule41(RuleDefinitionListIndexedBase):
+    """Rule 41 of ASHRAE 90.1-2019 Appendix G Section 5 (Envelope)"""
+
+    def __init__(self):
+        super(Section5Rule41, self).__init__(
+            rmrs_used=produce_ruleset_model_instance(
+                USER=False, BASELINE_0=True, PROPOSED=True
+            ),
+            each_rule=Section5Rule41.BuildingRule(),
+            index_rmr=BASELINE_0,
+            id="5-41",
+            description="Opaque roof surfaces that are not regulated (not part of opaque building envelope) must be "
+            "modeled with the same thermal emittance and solar reflectance in the baseline as in the "
+            "proposed design. ",
+            ruleset_section_title="Envelope",
+            standard_section="Section G3.1-5 Building Envelope Modeling Requirements for the Baseline building",
+            is_primary_rule=True,
+            list_path="ruleset_model_descriptions[0].buildings[*]",
+            data_items={"climate_zone": (BASELINE_0, "weather/climate_zone")},
+        )
+
+    class BuildingRule(RuleDefinitionListIndexedBase):
+        def __init__(self):
+            super(Section5Rule41.BuildingRule, self).__init__(
+                rmrs_used=produce_ruleset_model_instance(
+                    USER=False, BASELINE_0=True, PROPOSED=True
+                ),
+                each_rule=Section5Rule41.BuildingRule.RoofRule(),
+                index_rmr=BASELINE_0,
+                list_path="$.building_segments[*].zones[*].surfaces[*]",
+            )
+
+        def create_data(self, context, data=None):
+            building_b = context.BASELINE_0
+            return {
+                "scc_dict_b": get_surface_conditioning_category_dict(
+                    data["climate_zone"], building_b
+                ),
+            }
+
+        def list_filter(self, context_item, data=None):
+            surface_b = context_item.BASELINE_0
+            scc = data["scc_dict_b"][surface_b["id"]]
+            return (
+                get_opaque_surface_type(surface_b) == OST.ROOF
+                and scc is SCC.UNREGULATED
+            )
+
+        class RoofRule(RuleDefinitionBase):
+            def __init__(self):
+                super(Section5Rule41.BuildingRule.RoofRule, self).__init__(
+                    rmrs_used=produce_ruleset_model_instance(
+                        USER=False, BASELINE_0=True, PROPOSED=True
+                    ),
+                    required_fields={
+                        "$": ["optical_properties"],
+                    },
```

Comment:
I found this is the same as `absorptance_solar_exterior` or any of these data is missing. 
I suggest removing the required data field here.

Reply:
Addressed. Thank you for double-checking!